.github/workflows/scripts/e2e.generic.default.verify.sh

```diff
@@ -31,76 +32,75 @@ verify_provenance() {
     if [[ "$GITHUB_REF_TYPE" == "tag" ]]; then
         #TODO: try several versioned-tags and tags.
         SEMVER="$GITHUB_REF_NAME"
-        PATCH=$(echo "$SEMVER" | cut -d '.' -f3)
-        MINOR=$(echo "$SEMVER" | cut -d '.' -f2)
-        MAJOR=$(echo "$SEMVER" | cut -d '.' -f1)
-
-        M="${MAJOR:1}"
-        MAJOR_LESS_ONE="v$((M - 1))"
-        MINOR_LESS_ONE=$((MINOR - 1))
-        PATCH_LESS_ONE=$((PATCH - 1))
-        MAJOR_PLUS_ONE="v$((M + 1))"
-        MINOR_PLUS_ONE=$((MINOR + 1))
-        PATCH_PLUS_ONE=$((PATCH + 1))
+        MAJOR=$(version_major "$SEMVER")
+        MINOR=$(version_minor "$SEMVER")
+        PATCH=$(version_patch "$SEMVER")
+
+        MAJOR_LESS_ONE=$((${MAJOR:-1} - 1))
+        MINOR_LESS_ONE=$((${MINOR:-1} - 1))
+        PATCH_LESS_ONE=$((${PATCH:-1} - 1))
+        MAJOR_PLUS_ONE=$((${MAJOR:-0} + 1))
+        MINOR_PLUS_ONE=$((${MINOR:-0} + 1))
+        PATCH_PLUS_ONE=$((${PATCH:-0} + 1))
 
         # Correct vM.N.P
-        $verifier --branch "$BRANCH" --versioned-tag "$MAJOR.$MINOR.$PATCH" --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+        $verifier --branch "$BRANCH" --versioned-tag "v$MAJOR.$MINOR.$PATCH" --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
```

Comment:
in a follow-up PR, shall we crate a function for the verifier verification (since it's used by both the generic and go builders)?

Reply:
Yeah, maybe the `verify_provenance_authenticity` function you have for the Go test is good enough.

Reply:
ah yes, I created it sometimes after the comment. Lucky timing :-)